crates/sparrow-execution/src/tests.rs

```diff
@@ -1,263 +1,106 @@
-                channel.is_some(),
-                PipelineError::InputClosed {
-                    input,
-                    input_partition,
-                },
-            );
-            *channel = None;
-            Ok(())
-        }
-
-        fn do_work(
-            &self,
-            _partition: sparrow_scheduler::Partition,
-            _scheduler: &mut dyn sparrow_scheduler::Scheduler,
-        ) -> error_stack::Result<(), PipelineError> {
-            Ok(())
-        }
-    }
+    let source = Arc::new(sparrow_logical::Expr::new_uuid(
+        "read",
+        source_uuid,
+        source_type,
+        sparrow_logical::Grouping::new(0),
+    ));
+
+    let a_str = Arc::new(sparrow_logical::Expr::new_literal_str("a"));
+    let a =
+        Arc::new(sparrow_logical::Expr::try_new("fieldref", vec![source.clone(), a_str]).unwrap());
+    let b_str = Arc::new(sparrow_logical::Expr::new_literal_str("b"));
+    let b =
+        Arc::new(sparrow_logical::Expr::try_new("fieldref", vec![source.clone(), b_str]).unwrap());
+    let a_plus_b = Arc::new(sparrow_logical::Expr::try_new("add", vec![a, b]).unwrap());
+
+    let c_str = Arc::new(sparrow_logical::Expr::new_literal_str("c"));
+    let c =
+        Arc::new(sparrow_logical::Expr::try_new("fieldref", vec![source.clone(), c_str]).unwrap());
+    let a_plus_b_plus_c =
+        Arc::new(sparrow_logical::Expr::try_new("add", vec![a_plus_b.clone(), c]).unwrap());
+
+    let ab_str = Arc::new(sparrow_logical::Expr::new_literal_str("ab"));
+    let abc_str = Arc::new(sparrow_logical::Expr::new_literal_str("abc"));
+    let ab_and_abc = Arc::new(
+        sparrow_logical::Expr::try_new("record", vec![ab_str, a_plus_b, abc_str, a_plus_b_plus_c])
+            .unwrap(),
+    );
+
+    let plan = sparrow_backend::compile(&ab_and_abc, None).unwrap();
+
+    let (output_tx, mut output_rx) = tokio::sync::mpsc::channel(10);
+
+    let source = sparrow_sources::InMemory::new(true, source_schema.clone()).unwrap();
+
+    // Before running, make sure there is an input batch available.
+    let time = Arc::new(TimestampNanosecondArray::from(vec![0, 1, 2, 3]));
+    let subsort = Arc::new(UInt64Array::from(vec![0, 1, 2, 3]));
+    let key_hash = Arc::new(UInt64Array::from(vec![0, 1, 2, 3]));
+    let a = Arc::new(Int64Array::from(vec![0, 1, 2, 3]));
+    let b = Arc::new(Int64Array::from(vec![4, 7, 10, 11]));
+    let c = Arc::new(Int64Array::from(vec![Some(21), None, Some(387), Some(87)]));
+    let input_batch = RecordBatch::try_new(
+        source_prepared_schema.clone(),
+        vec![time, subsort, key_hash, a, b, c],
+    )
+    .unwrap();
+    session.rt.block_on(source.add_batch(input_batch)).unwrap();
+
+    // Create the exececutor after the data is added.
+    // Currently, we snapshot the data when performing a non-live query.
+    //
+    // TODO: Support live query (via configuration).
+    let executor = {
+        let mut sources: HashMap<_, &dyn Source> = HashMap::with_capacity(1);
```

Comment:
Having to make this hashmap to pass to the execution was a little gross, since we'll likely need to have a `HashMap<_, Arc<dyn Source>>` else where for the sources we know about. Figuring out how this interacts with Python is probably a good follow-up.